src/peer-discovery/tests/index.js

```diff
@@ -30,5 +44,49 @@ module.exports = (common) => {
       await discovery.start()
       await discovery.start()
     })
+
+    it('should listen a peer event after start', async () => {
+      const defer = pDefer()
+      await discovery.start()
+
+      discovery.once('peer', ({ id, multiaddrs }) => {
+        expect(id).to.exist()
+        expect(PeerId.isPeerId(id)).to.eql(true)
+        expect(multiaddrs).to.exist()
+
+        multiaddrs.forEach((m) => expect(multiaddr.isMultiaddr(m)).to.eql(true))
+
+        defer.resolve()
+      })
+
+      return defer.promise
```

Comment:
Avoid mixing async and promise usage

```suggestion
      await defer.promise
```